frontend/src/components/PetitionCard.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  Dispatch,
+  FormEventHandler,
+  ReactNode,
+  SetStateAction,
+  useEffect,
+  useState,
+} from "react";
+import { FaSpinner } from "react-icons/fa";
+import { PetitionDetail, Question } from "../models/Petition";
+import { submitPetitionSignature } from "../utils/backend";
+
+interface Props {
+  className?: string;
+  petition: PetitionDetail;
+  link?: ReactNode;
+}
+
+enum State {
+  Initial,
+  Submitting,
+  Error,
+  Existed,
+  Success,
+}
+
+const SESSION_KEY = "petitionData";
+
+function makeBox(
+  props: Props,
+  children: ReactNode,
+  onsubmit: FormEventHandler<HTMLFormElement> | undefined = undefined
+) {
+  return (
+    <div className={props.className}>
+      <form
+        className="bg-primary rounded-2xl p-4 text-white flex flex-col gap-2"
+        onSubmit={onsubmit}
+      >
+        {children}
+      </form>
+    </div>
+  );
```

Comment:
Momenteel hebben we altijd gehanteerd dat er geen haakjes om een HTML blok te hoeven staan in TSX. Dus direct de <div> teruggeven

Reply:
De haakjes zijn vanwege formatting die Prettier automatisch doet. Anders mag er geen enter direct na de return statement